odoo/addons/base/models/ir_model.py

```diff
@@ -1136,15 +1136,12 @@ def _update_selection(self, model_name, field_name, selection, ondelete):
 
         # selection rows {value: row}
         cur_rows = self._existing_selection_data(model_name, field_name)
-        new_rows = {}
-        for index, (value, label) in enumerate(selection):
-            ondelete_opt = ondelete.get(value, 'null')
-            new_rows[value] = {
-                'value': value,
-                'name': label,
-                'sequence': index,
-                'ondelete': ondelete_opt if not callable(ondelete_opt) else 'callable',
-            }
+        new_rows = {
+            value: dict(value=value, name=label, sequence=index,
+                        ondelete='callable' if callable(policy) else policy)
+            for index, (value, label) in enumerate(selection)
+            for policy in [ondelete.get(value, 'null')]
+        }
```

Comment:
I'm okay with all the other changes, but multi-line list / dict / set comprehensions are my pet peeve, they can be idiomatic and elegant when used sparingly but here they have no added value, they make the code less readable and you even have to go out of your way and iterate over a single-item list just to define the variable `policy` so that you can use it in the ternary assignment expression above

Reply:
okay, gonna change it